Analysis/Tasks/PWGCF/dptdptcorrelations-simchl.cxx

```diff
@@ -0,0 +1,927 @@
+    /* versus centrality/multiplicity  profiles */
+    TProfile* fhN1_vsC[2];           //!<! weighted single particle distribution vs event centrality, track 1 and 2
+    TProfile* fhSum1Pt_vsC[2];       //!<! accumulated sum of weighted \f$p_T\f$ vs event centrality, track 1 and 2
+    TProfile* fhN1nw_vsC[2];         //!<! un-weighted single particle distribution vs event centrality, track 1 and 2
+    TProfile* fhSum1Ptnw_vsC[2];     //!<! accumulated sum of un-weighted \f$p_T\f$ vs event centrality, track 1 and 2
+    TProfile* fhN2_vsC[4];           //!<! weighted accumulated two particle distribution vs event centrality 1-1,1-2,2-1,2-2, combinations
+    TProfile* fhSum2PtPt_vsC[4];     //!<! weighted accumulated \f${p_T}_1 {p_T}_2\f$ distribution vs event centrality 1-1,1-2,2-1,2-2, combinations
+    TProfile* fhSum2DptDpt_vsC[4];   //!<! weighted accumulated \f$\sum ({p_T}_1- <{p_T}_1>) ({p_T}_2 - <{p_T}_2>) \f$ distribution vs event centrality 1-1,1-2,2-1,2-2, combinations
+    TProfile* fhN2nw_vsC[4];         //!<! un-weighted accumulated two particle distribution vs event centrality 1-1,1-2,2-1,2-2, combinations
+    TProfile* fhSum2PtPtnw_vsC[4];   //!<! un-weighted accumulated \f${p_T}_1 {p_T}_2\f$ distribution vs event centrality 1-1,1-2,2-1,2-2, combinations
+    TProfile* fhSum2DptDptnw_vsC[4]; //!<! un-weighted accumulated \f$\sum ({p_T}_1- <{p_T}_1>) ({p_T}_2 - <{p_T}_2>) \f$ distribution vs \f$\Delta\eta,\;\Delta\phi\f$ distribution vs event centrality 1-1,1-2,2-1,2-2, combinations
+
+    const char* tname[2] = {"1", "2"}; ///< the external track names, one and two, for histogram creation
+    const char* trackPairsNames[4] = {"OO", "OT", "TO", "TT"};
+
+    /// \brief Returns the potentially phi origin shifted phi
+    /// \param phi the track azimuthal angle
+    /// \return the track phi origin shifted azimuthal angle
+    inline float GetShiftedPhi(float phi)
+    {
+      if (not(phi < phiup)) {
+        return phi - M_PI * 2;
+      } else {
+        return phi;
+      }
+    }
+
+    /// \brief Returns the zero based bin index of the eta phi passed track
+    /// \param t the intended track
+    /// \return the zero based bin index
+    ///
+    /// According to the bining structure, to the track eta will correspond
+    /// a zero based bin index and similarlly for the track phi
+    /// The returned index is the composition of both considering eta as
+    /// the first index component
+    /// WARNING: for performance reasons no checks are done about the consistency
+    /// of track's eta and phin with the corresponding ranges so, it is suppossed
+    /// the track has been accepted and it is within that ranges
+    /// IF THAT IS NOT THE CASE THE ROUTINE WILL PRODUCE NONSENSE RESULTS
+    template <typename TrackObject>
+    inline int GetEtaPhiIndex(TrackObject const& t)
```

Comment:
Same comment about the inline here. If they are not needed, i suggest to remove them. It makes the code easier and more readable.

Reply:
You were right! https://isocpp.org/wiki/faq/inline-functions
I will update the PR accordingly